internal/graph/lookupresources.go

```diff
@@ -36,6 +33,10 @@ type ValidatedLookupResourcesRequest struct {
 	Revision datastore.Revision
 }
 
+// reachableResourcesLimit is a limit set on the reachable resources calls to ensure caching
+// stores smaller chunks.
+const reachableResourcesLimit = 1000
```

Comment:
Actually, no: we're not forwarding the limit because of the filtering aspect: a request for 100 resources from LR might need 1000 from reachable resources, because 900 are checked and are not returned. This is, to be fair, a corner case and in *most* cases the limits should be fairly evenly matched, but it is possible, so we run a (small) risk of having to make extra dispatch requests to reachable resources in that scenario. Thoughts?

Reply:
are you implying by omission that caching is not a problem and that different RR response sizes are reusable regardless?

>Actually, no: we're not forwarding the limit because of the filtering aspect: a request for 100 resources from LR might need 1000 from reachable resources, because 900 are checked and are not returned. This is, to be fair, a corner case and in most cases the limits should be fairly evenly matched, but it is possible, so we run a (small) risk of having to make extra dispatch requests to reachable resources in that scenario. Thoughts?

I think this once again one of those situations where the system should adapt dynamically keeping stats around, and issue the limit that yields less amount of dispatch requests. For the time being this is fine, but let's open follow up issues because it will be forgotten in the ReachableResources ocean 🙏🏻